lib/inferno/dsl/fhir_resource_validation.rb

```diff
@@ -277,16 +277,15 @@ def call_validator(resource, profile_url)
         end
 
         # @private
-        def operation_outcome_from_hl7_wrapped_response(response)
-          res = JSON.parse(response)
-          if res['sessionId'] != @session_id
-            validator_session_repo.save(test_suite_id:, validator_session_id: res['sessionId'],
+        def operation_outcome_from_hl7_wrapped_response(response_hash)
+          if response_hash['sessionId'] != @session_id
```

Comment:
Sorry for scope creep, but this specific line is the focus of a separate bug (FI-2914 "Don't try to store null validator session IDs") . Can you also add a null check on `response_hash['sessionId']` while we're looking at this file? Something like this is probably enough to resolve that ticket too
```suggestion
          if response_hash['sessionId'] and response_hash['sessionId'] != @session_id
```